docs/proposals/perf.md

```diff
@@ -0,0 +1,231 @@
+
+1. A real K8S Cluster that has master and nodes.
+2. A separated VM where the KubeEdge Cloud Part Services are running, including Edge Controller and Cloud Hub and so on.
+3. Test Client(E2E) will build KubeEdge EdgeNode image and put into the Docker Hub repository.
+4. Test Client(E2E) uses the deployment controller to deploy KubeEdge EdgeNodes as pods with replicas.
+
+When using the KubeEdge Performance Test, the developer is responsible for creating #1 and #2 above.
+
+#### K8S Master
+| Subject                        | Description                                  |
+|--------------------------------|----------------------------------------------|
+| OS                             |  Ubuntu 18.04 server 64bit                   |
+| K8S Version                    |  v1.13.5                                     |
+| Docker Version                 |  v17.09                                      |
+| CPU                            |  32vCPUs                                     |
+| RAM                            |  128GB                                       |
+| Disk Size                      |  40GB                                        |
+| Count                          |  1                                           |
+
+This VM is used to run K8S Master Services including K8S API Server and K8S Scheduler and so on.
+
+#### KubeEdge Cloud Part
+| Subject                        | Description                                  |
+|--------------------------------|----------------------------------------------|
+| OS                             |  Ubuntu 18.04 server 64bit                   |
+| Docker Version                 |  v17.09                                      |
+| CPU                            |  32vCPUs                                     |
+| RAM                            |  128GB                                       |
+| Disk Size                      |  40GB                                        |
+| Count                          |  1                                           |
+
+This VM is used to run KubeEdge Cloud Part Services including Edge Controller and Cloud Hub and so on.
+
+#### K8S Nodes
+| Subject                        | Description                                  |
+|--------------------------------|----------------------------------------------|
+| OS                             |  Ubuntu 18.04 server 64bit                   |
+| Docker Version                 |  v17.09                                      |
+| CPU                            |  32vCPUs                                     |
+| RAM                            |  128GB                                       |
+| Disk Size                      |  40GB                                        |
+| Count                          |  1...N                                       |
+
+These VMs are used to deploy numbers of KubeEdge Edge Nodes pods which are running Edged and EdgeHub and so on.
```

Comment:
will we have 1 vm per edge node pod ? I thought the concept of hollow node was to avoid have a large number of vm's as we are just running hollow edge node pods which are mimicking a real vm .

Reply:
@rohitsardesai83, no, we will have lots of edge node pods in 1 vm as the deployment graph.
But we need to consider the vcpu and ram of the vm. because 1 vm is not enough to run all of the edge node pods. We will clarify it by some notes

Reply:
@rohitsardesai83 , for KubeEdge perofrmance setup is similar to kubemark setup, where they simulate no.of hollow-nodes in pods and deploy on k8s cluster. in kubeedge also we do the similar kind of simulation for creating kubeedge node in pod and deploy the pods through deploment object.

Reply:
@rohitsardesai83 more details are added in the lasest commit with @pavan187